http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/ChannelLogHelper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal.utils;
+
+import io.netty.channel.Channel;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * Utility class for logging the channel ID along with the message when within the context of a channel.
+ */
+@SdkInternalApi
+public final class ChannelLogHelper {
```

Comment:
This class LGTM, but we could also consider making it object-based rather than utility-based, so that calls don't need to repeatedly include a `logger` argument, and to reduce the likelihood of using a direct logger call.